otcextensions/sdk/sdrs/v1/_proxy.py

```diff
@@ -10,8 +10,799 @@
+        :param str production_flavor: flavor ID for production site
+            If 'flavor' is specified this parameter doesn't take effect
+        :param str dr_flavor: flavor ID for DR site
+            If 'flavor' is specified this parameter doesn't take effect
+        """
+        protected_instance = self._get_resource(
+            _p_instance.ProtectedInstance,
+            protected_instance)
+        return protected_instance.modify_instance(
+            self,
+            protected_instance=protected_instance.id,
+            flavor=flavor,
+            production_flavor=production_flavor,
+            dr_flavor=dr_flavor
+        )
+
+    # ======== Replication pair ========
+
+    def create_replication_pair(self, **attrs):
+        """Create a replication pair using attributes
+
+        :param dict attrs: Keyword arguments which will be used to create
+            a :class:`~otcextensions.sdk.sdrs.v1.replication_pair.
+            ReplicationPair`, comprised of the properties on the Replication
+            Pair class.
+        :returns: The results of config creation
+        :rtype: :class:`~otcextensions.sdk.sdrs.v1.replication_pair.
+            ReplicationPair`
+        """
+        return self._create(
+            _replication_pair.ReplicationPair,
+            **attrs
+        )
+
+    def delete_replication_pair(self,
+                                replication,
+                                server_group_id=None,
+                                delete_target_volume=False,
+                                ignore_missing=True):
+        """Delete a single SDRS replication pair
+
+        :param replication: The value can be the ID of a replication pair
+             or a :class:
+             `~otcextensions.sdk.sdrs.v1.replication_pair.ReplicationPair`
+             instance.
+        :param bool server_group_id: Protection group ID of replication pair
+        :param bool delete_target_volume: Specifies whether DR site
+            volume should be deleted after replication pair deletion
+        :param bool ignore_missing: When set to ``False``
+            :class:`~openstack.exceptions.ResourceNotFound` will be raised
+            when the group does not exist.
+            When set to ``True``, no exception will be set when attempting to
+            delete a nonexistent protected instance
+        :returns: instance of
+            :class:
+            `~otcextensions.sdk.sdrs.v1.replication_pair.ReplicationPair`
+        """
+        res = self._get_resource(_replication_pair.ReplicationPair,
+                                 replication)
+        try:
+            del_in = res.delete(self, server_group_id=server_group_id,
+                                delete_target_volume=delete_target_volume)
+        except exceptions.ResourceNotFound:
+            if ignore_missing:
+                return None
+            raise
+        return del_in
```

Comment:
try to move this into resource method

Reply:
moved try-except block inside resource methods for 'delete_protected_instance' and 'delete_replication_pair' methods